src/scripts/clipperUI/clipper.tsx

```diff
@@ -620,9 +621,22 @@ class ClipperClass extends ComponentBase<ClipperState, {}> {
 
 		this.state.setState({ oneNoteApiResult: { status: Status.InProgress } });
 		SaveToOneNote.startClip(this.state).then((startClipPackage: StartClipPackage) => {
-			clipEvent.setCustomProperty(Log.PropertyName.Custom.CorrelationId, startClipPackage.responsePackage.request.getResponseHeader(Constants.HeaderValues.correlationId));
-			clipEvent.setCustomProperty(Log.PropertyName.Custom.AnnotationAdded, startClipPackage.annotationAdded);
-			this.state.setState({ oneNoteApiResult: { data: startClipPackage.responsePackage.parsedResponse, status: Status.Succeeded } });
+			RatingsHelper.incrementClipSuccessCount().then(() => {
+				// do nothing special on incrementClipSuccessCount resolve
+			}).then(() => {
+				// will happen regardless of success/failure of incrementClipSuccessCount
```

Comment:
yep, which is why I don't have a reject handler. but if something unexpectedly throws inside of the method (granted, unlikely given the method), it will reject. so I left this empty resolve handler around just to ensure that the next `.then` _definitely_ always fires. maybe there's a cleaner way of ensuring this.

does that answer what I think you're asking?